test/integration/integration_suite_test.go

```diff
@@ -270,3 +270,39 @@ func getRegionsWithCapsAndPlans(t *testing.T, client *linodego.Client, capabilit
 
 	return result
 }
+
+// getRegionsWithCapsAndSiteType returns a list of regions that meet the given capabilities and site type
+func getRegionsWithCapsAndSiteType(t *testing.T, client *linodego.Client, capabilities []string, siteType string) []string {
+	result := make([]string, 0)
+
+	regions, err := client.ListRegions(context.Background(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	regionHasCaps := func(r linodego.Region) bool {
+		capsMap := make(map[string]bool)
+
+		for _, c := range r.Capabilities {
+			capsMap[strings.ToUpper(c)] = true
+		}
+
+		for _, c := range capabilities {
+			if _, ok := capsMap[strings.ToUpper(c)]; !ok {
+				return false
+			}
+		}
+
+		return true
+	}
```

Comment:
optional: Would it make sense to call `getRegionsWithCaps(...)` here and filter down the results below?

Reply:
I did try to call `getRegionsWithCaps(...)` at first but figured out it might be better to have the logic here. Because `getRegionsWithCaps(...)` only return the region's id string, we will have to make extra API calls to get the Region object for obtaining necessary info. It's probably slow down the test. 

I'm wondering if it makes sense to move `regionHasCaps(...)` as an independent function, so that we can call it here without repeating this part of logic. What's your thoughts?

Reply:
I think that would be a good idea. Maybe we can move it as a separate helper function in `integration_test_suite.go`

Reply:
@ykim-1 @lgarber-akamai thanks for the comment! Just applied the improvement, could both of you re-approve the PR?